### PR TITLE
completed getCommentsByArticleId

### PR DESCRIPTION
**Pull request by Superjim**
I have moved my error handling to a promise reject in the model, which is passed to my error handler in app.js

I have completed the getCommentsByArticleId functionality

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -152,20 +152,19 @@ describe("nc-news", () => {
...
-    test("responds with an array of comments sorted by date with most recent comments first", () => {
+    test("responds with an array of comment objects sorted by date with most recent comments first", () => {
       return request(app)
         .get("/api/articles/1/comments")
         .expect(200)
         .then((response) => {
-          const comments = response.body.articles;
+          const comments = response.body.comments;
           expect(Array.isArray(comments)).toBe(true);
           expect(comments.length > 0).toBe(true);
```
Comment (xViolaine) [inline __tests__/app.test.js:169]: I would probably not even just test whether the comments are longer than 0, I would probably look for the exact length as we know how many comments there are!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -152,20 +152,19 @@ describe("nc-news", () => {
...
           //Date parse the created_at property and check its greater than the next one along the array. Run to array.length - 1 so it doesnt compare undefined.
```
Comment (xViolaine) [inline __tests__/app.test.js:170]: Nice job thinking of this solution, but we do have a very nice jest-sorted package at our disposal for these sort of things - makes life a lot easier!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -152,20 +152,19 @@ describe("nc-news", () => {
...
-          expect(comment[0].created_at).toBe(1);
-          expect(comment[0].author).toBe(1);
-          expect(comment[0].body).toBe(1);
-          expect(comment[0].article_id).toBe(1);
+          expect(allComments[0].comment_id).toBe(5);
+          expect(allComments[0].votes).toBe(0);
+          expect(allComments[0].author).toBe("icellusedkars");
+          expect(allComments[0].body).toBe("I hate streaming noses");
+          expect(allComments[0].article_id).toBe(1);
         });
```
Comment (xViolaine) [inline __tests__/app.test.js:160]: We also want a 200 test for when the ID is valid, but it has no comments - should come back with an empty array then

**Review by xViolaine — CHANGES_REQUESTED** [review]
